lib/maze/request_repeater.rb

```diff
@@ -0,0 +1,48 @@
+module Maze
+  # Repeats POST requests
+  module RequestRepeater
+
+    def do_POST(request, response)
+      repeat(request) if enabled?
+
+      super(request, response)
+    end
+
+    private
+
+    def enabled?
+      # enabled if the config option is on and this request type should be repeated
+      Maze.config.repeater_api_key && url_for_request_type
+    end
+
+    # @param request [HTTPRequest] The request to be repeated
+    def repeat(request)
+
+      # TODO: Forwarding of internal errors to be considered later
+      return if request.header.keys.any? { |key| key.downcase == 'bugsnag-internal-error' }
+
+      url = url_for_request_type
+      http = Net::HTTP.new(url.host)
+      bugsnag_request = Net::HTTP::Post.new(url.path)
+      bugsnag_request['Content-Type'] = 'application/json'
+
+      # Set all Bugsnag headers that are present
+      bugsnag_request.body = request.body
+      request.header.each {|key,value| bugsnag_request[key] = value if key.downcase.start_with? 'bugsnag-' }
+      bugsnag_request['bugsnag-api-key'] = Maze.config.repeater_api_key
+
+      # TODO Also overwrite apiKey in the payload, if present, and recalculate the integrity header
```

Comment:
How much of an issue is this / do we know what the pipeline does if the API keys are different?

I wonder if we should make this an error case so that something unexpected doesn't happen, though I guess the worst-case is that this uses the API key in the payload and gets dropped since it doesn't match a project

Reply:
I spoke to the pipeline team about it and they said the one in the body is legacy and not inspected.  So the change here would be just be for consistency's sake.  We wouldn't want to make it and error case as some notifiers still set both the header and value in the body (consistently, but will be a dummy key in test scenarios).